specification/managementgroups/resource-manager/Microsoft.Management/stable/2020-02-01/management.JSON

```diff
@@ -0,0 +1,2000 @@
+        },
+        "properties": {
+          "title": "Properties",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ManagementGroupInfoProperties"
+        }
+      }
+    },
+    "HierarchySettings": {
+      "description": "Settings defined at the Management Group scope.",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified ID for the settings object.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000/settings/default.",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the resource.  For example, Microsoft.Management/managementGroups/settings.",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the object. In this case, default.",
+          "readOnly": true
+        },
+        "properties": {
+          "title": "Properties",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/HierarchySettingsProperties"
+        }
+      }
+    },
+    "HierarchySettingsProperties": {
+      "description": "The generic properties of hierarchy settings.",
+      "type": "object",
+      "properties": {
+        "tenantId": {
+          "type": "string",
+          "description": "The AAD Tenant ID associated with the hierarchy settings. For example, 00000000-0000-0000-0000-000000000000"
+        },
+        "requireAuthorizationForGroupCreation": {
+          "type": "boolean",
+          "description": "Indicates whether RBAC access is required upon group creation under the root Management Group. If set to true, user will require Microsoft.Management/managementGroups/write action on the root Management Group scope in order to create new Groups directly under the root. This will prevent new users from creating new Management Groups, unless they are given access."
+        },
+        "defaultManagementGroup": {
+          "type": "string",
+          "description": "Settings that sets the default Management Group under which new subscriptions get added in this tenant. For example, /providers/Microsoft.Management/managementGroups/defaultGroup"
+        }
+      }
+    },
+    "HierarchySettingsList": {
+      "description": "Lists all hierarchy settings.",
+      "properties": {
+        "value": {
+          "description": "The list of hierarchy settings.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/HierarchySettingsInfo"
+          }
+        },
+        "nextLink": {
+          "description": "The URL to use for getting the next set of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "HierarchySettingsInfo": {
+      "description": "The hierarchy settings resource.",
+      "properties": {
+        "id": {
```

Comment:
You should reference the common types to bring in 'id', 'name', 'type'...

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json

Reply:
As Ryan said, you can reduce this verbosity by just reffing common types :-)

---
In reply to: [385316007](https://github.com/Azure/azure-rest-api-specs/pull/8443#discussion_r385316007) [](ancestors = 385316007)